### ISSUE TITLE
amp-carousel jumps to last slide when the visibility of its parent container is toggled

### ISSUE THREAD

**Issue by ruborg** [ampproject/amphtml#9882]
## What's the issue?

amp-carousel jumps to last slide when the visibility of its parent container is toggled

## How do we reproduce the issue?

See jsbin here:
http://jsbin.com/zosomuqezo/edit?html,output

1.Tap blue or green preview thumbnail (i.e. not first or last)
2. Tap carousel 1
3. Tap carousel 0
4. Carousel 0 now jumps to final slide

**Expected**: carousel 0 stays on same slide.

Note that the problem does not occur if the visibility of the carousel is toggled directly, as opposed to its parent, e.g. in this jsbin the issue **does not occur**
http://jsbin.com/girekiyepa/edit?html,output

(Note, the problem also occurs if amp-bind is used to interact with the carousel & trigger [slide] and [class] changes (took me a couple of days to realise the issue wasn;t related to amp-bind :'(    )

## What browsers are affected?

Firefox 53.0.3, 54.0 MacOS Desktop FAIL, Firefox mobile Android FAIL
Chrome desktop OK, Chrome mobile Android OK, Firefox mobile iOS10 OK

That's all I checked

## Which AMP version is affected?

1496877433269

**Comment by camelburrito** [comment]
@ruborg : Does this happen on a specific OS/Browser/version?

**Comment by camelburrito** [comment]
WoW! @ruborg this is a great find - Thanks for filing!

CC @jridgewell, @dvoytenko  - This is super interesting , i have no clue why this is happening! 

@jridgewell can you think of why something like this would happen

**Comment by aghassemi** [comment]
lower priority since looks like it is FF only.

**Comment by dvoytenko** [comment]
@camelburrito Why this happens seems pretty obvious to me. It appears that FF fires a scroll event on carousel when it changes `display` style. It looks like in the case when `display: none`, the container width is measured as 0 and thus it calculates that the last slide must be visible.

Not sure what's the best workaround for this. I'd try to avoid responding to scroll events due to display changes. Not sure if that's possible. Maybe we can just ignore scroll events when `slideWidth_ == 0`?

**Comment by dvoytenko** [comment]
I'm changing this back to P1 - I think there's a chance that other browsers would behave similarly.

**Comment by camelburrito** [comment]
oh yeah this makes total sense now, i tested this and it was only replicable on FF, not sure is this is still a P1, i guess we will have to measure the width of the container and only proceed when width is not 0. 

Does onLayoutMeasure fire in such scenarios?

**Comment by dvoytenko** [comment]
Yes, `onLayoutMeasure` does fire for `display:none`, however, it's no guarantee that it will fire immediately - most of times it'll be somewhat delayed (async process). My guess in this scenario it happens a bit faster than in others.

**Comment by ampprojectbot** [comment]
This is a high priority issue but it hasn't been updated in awhile. @camelburrito Do you have any updates?

**Comment by ampprojectbot** [comment]
This is a high priority issue but it hasn't been updated in awhile. @camelburrito Do you have any updates?

**Comment by ampprojectbot** [comment]
This is a high priority issue but it hasn't been updated in awhile. @camelburrito Do you have any updates?

**Comment by ampprojectbot** [comment]
This is a high priority issue but it hasn't been updated in awhile. @camelburrito Do you have any updates?

**Comment by ampprojectbot** [comment]
This is a high priority issue but it hasn't been updated in awhile. @camelburrito Do you have any updates?

**Comment by ampprojectbot** [comment]
This is a high priority issue but it hasn't been updated in awhile. @camelburrito Do you have any updates?

**Comment by ampprojectbot** [comment]
This is a high priority issue but it hasn't been updated in awhile. @camelburrito Do you have any updates?

**Comment by ampprojectbot** [comment]
This is a high priority issue but it hasn't been updated in awhile. @aghassemi Do you have any updates?

**Comment by ampprojectbot** [comment]
This is a high priority issue but it hasn't been updated in awhile. @aghassemi Do you have any updates?

**Comment by ampprojectbot** [comment]
This is a high priority issue but it hasn't been updated in awhile. @aghassemi Do you have any updates?

**Comment by ampprojectbot** [comment]
This is a high priority issue but it hasn't been updated in awhile. @aghassemi Do you have any updates?

**Comment by nainar** [comment]
This issue is fixed for all users who upgrade their `<amp-carousel>` version to 0.2. 

For more details please see documentation here: https://github.com/ampproject/amphtml/blob/master/extensions/amp-carousel/amp-carousel.md
